base/applications/utilman/dialog.c

```diff
@@ -152,74 +158,79 @@ INT ListBoxRefreshContents(VOID)
     /* Remove all the items from the listbox */
     SendMessageW(Globals.hListDlg, LB_RESETCONTENT, 0, 0);
 
-    if (!IsProcessRunning(EntriesList[0].lpProgram))
+    if (Globals.iArrayIndex == 0)
```

Comment:
To help you understand why the current code is not good, try thinking how you would do that if you had to support (let's say) 10 or more accessibility tools, or more accurately an arbitrary number of them (as it appears to be the case on Windows, since their list is stored in the registry and not hardcoded in osk as it is done here).